src/button.cpp

```diff
@@ -0,0 +1,27 @@
+#include "button.h"
+
+Button::Button(uint8_t pin) : input_pin_(pin) {
+  pinMode(pin, INPUT_PULLUP);
+}
+
+// Returns either "NEUTRAL", "PUSH" or "HOLDING"
+// It is advised to treat "HOLDING" the same as "NEUTRAL" outside this function
+ButtonState Button::getState() {
+  ButtonState state;
+  if (digitalRead(this->input_pin_) == LOW) {
+    if (this->last_position_ != ButtonState::PUSH ||
+        (millis() - this->last_print_time_) > this->hold_time_) {
+      this->last_print_time_ = millis();
+      this->last_position_ = ButtonState::PUSH;
+      state = ButtonState::PUSH;
+    } else {
+      state = ButtonState::HOLDING;
+    }
+  } else {
+    this->last_position_ = ButtonState::NEUTRAL;
+    state = ButtonState::NEUTRAL;
+  }
+  // To prevent multiple 'up's and 'down's per press due to bouncing
+  usleep(this->bounce_time_);
```

Comment:
Pins in arduino can be declared with a bounce using a certain library; https://playground.arduino.cc/Code/Bounce/. It's been a while since I have used this but if i remember correctly it is pretty good to use.

Reply:
Nice it will note it and change it later, since it is now out of scope